components/ws-manager/pkg/manager/monitor.go

```diff
@@ -1254,6 +1254,17 @@ func (m *Monitor) finalizeWorkspaceContent(ctx context.Context, wso *workspaceOb
 		return true, gitStatus, err
 	}
 
+	handleGRPCError := func(ctx context.Context, err error) error {
```

Comment:
This will make code harder to read.
We already have function `handleGRPCError` here: 
https://github.com/gitpod-io/gitpod/blob/a257b53a17f7fc0b9f9ea36d28f9ccc6f6087958/components/ws-manager/pkg/manager/monitor.go#L1414

So overloading it here with an inline function will be confusing later on.

Can we instead check if err is Unavailable prior to calling handleGRPCError and handle that case?

Reply:
fixed